### PR TITLE
new_audit: add js-libraries audit, just listing detected js libs

### PR DESCRIPTION
**Pull request by paulirish**
cc @housseindjirdeh 

output in the LHR looks like this:

<details>

```json
{
  "id": "js-libraries",
  "title": "JavaScript libraries",
  "description": "All detected front-end JavaScript libraries on the page",
  "score": 1,
  "scoreDisplayMode": "binary",
  "rawValue": true,
  "details": {
    "type": "table",
    "headings": [
      {
        "key": "name",
        "itemType": "text",
        "text": "Name"
      },
      {
        "key": "version",
        "itemType": "text",
        "text": "Version"
      }
    ],
    "items": [
      {
        "name": "jQuery",
        "version": "1.12.4",
        "npm": "jquery"
      },
      {
        "name": "jQuery UI",
        "version": "1.11.4",
        "npm": "jquery-ui"
      },
      {
        "name": "Modernizr",
        "version": "2.6.2",
        "npm": "modernizr"
      },
      {
        "name": "Underscore",
        "version": "1.8.3",
        "npm": "underscore"
      },
      {
        "name": "FlexSlider",
        "version": null,
        "npm": "flexslider"
      }
    ],
    "summary": {}
  }
}
```

</details>

### CODE REVIEW

**Comment by paulirish** [comment]
cc @rviscomi to review the data format

**Comment by developit** [comment]
Looks good! This should be relatively easy to parse from within BQ.

@rviscomi - any thoughts about whether this format makes it possible to flatten the list of frameworks down so it's more readily accessible (as columns)?

**Comment by patrickhulce** [comment]
we should also probably bump our version of the detector, last time I checked they were missing ~90% of react sites I tested and a few commits helped lower that significantly

**Comment by wardpeet** [comment]
this is awesome :)

**Comment by rviscomi** [comment]
> any thoughts about whether this format makes it possible to flatten the list of frameworks down so it's more readily accessible (as columns)?

Yeah with a user-generated function we're able to massage this formatting however we need. This format LGTM.

**Comment by paulirish** [comment]
> we should also probably bump our version of the detector, last time I checked they were missing ~90% of react sites I tested and a few commits helped lower that significantly

yup. already on it: https://github.com/johnmichel/Library-Detector-for-Chrome/issues/118

File: lighthouse-core/audits/dobetterweb/js-libraries.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (brendankenny) [inline lighthouse-core/audits/dobetterweb/js-libraries.js]: 2018

File: lighthouse-core/test/audits/dobetterweb/js-libraries-test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (brendankenny) [inline lighthouse-core/test/audits/dobetterweb/js-libraries-test.js]: 2018

File: lighthouse-core/test/audits/dobetterweb/js-libraries-test.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/* eslint-env jest */
+describe('Returns detected front-end JavaScript libraries', () => {
+  it('always passes', () => {
+    // no libraries
+    const auditResult1 = JsLibrariesAudit.audit({
+      JSLibraries: [],
+    });
+    assert.equal(auditResult1.rawValue, true);
+
+    // duplicates. TODO: consider failing in this case
```
Comment (brendankenny) [inline lighthouse-core/test/audits/dobetterweb/js-libraries-test.js]: > TODO: consider failing in this case

it's not *that* uncommon to have more than one jquery in one page. And if we merge network stuff from iframes, won't it be even more common?

File: typings/audit.d.ts [hunk 4]
```diff
@@ -87,7 +87,7 @@ declare global {
 
     export type DetailsItem = string | number | DetailsRendererNodeDetailsJSON |
       DetailsRendererLinkDetailsJSON | DetailsRendererCodeDetailJSON | undefined |
-      boolean | DetailsRendererUrlDetailsJSON;
+      boolean | DetailsRendererUrlDetailsJSON | null;
```
Comment (brendankenny) [inline typings/audit.d.ts]: details make me sad

File: lighthouse-core/audits/dobetterweb/js-libraries.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+class JsLibrariesAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'js-libraries',
+      title: 'Detected JavaScript libraries',
+      description: 'All front-end JavaScript libraries detected on the page.',
```
Comment (exterkamp) [inline lighthouse-core/audits/dobetterweb/js-libraries.js]: Does this need i18n?

**Pull request merged**